gateway/mw_oauth2_auth.go

```diff
@@ -24,12 +25,25 @@ const (
 )
 
 type OAuthHeaderProvider interface {
-	getOAuthHeaderValue(r *http.Request, OAuthSpec *UpstreamOAuth) (string, error)
+	getOAuthToken(r *http.Request, OAuthSpec *UpstreamOAuth) (string, error)
 }
 
-type DistributedCacheOAuthProvider struct{}
+type ClientCredentialsOAuthProvider struct{}
 
-type PerAPIOAuthProvider struct{}
+type PerAPIClientCredentialsOAuthProvider struct{}
```

Comment:
@andrei-tyk is this being used anywhere now?

Reply:
currently not, it is there to be reused when we'll decide to have the "token per gateway" functionality available, it's already there and it's just a renaming